File: src/ts/math/pow.ts

// https://www.khanacademy.org/computing/computer-science/algorithms/recursive-algorithms/a/computing-powers-of-a-number
export default function pow (x: number, n: number):number {
  // initial condition
  if (n === 1) {
    return x
  }

  // even number
  if ((n % 2) === 0) {
    return pow(x * x, n / 2)
  }

  // odd number
  return x * pow(x * x, (n -1) / 2)
}

File: src/go/math/pow.go
package math

// Pow of x n
func Pow(x int, n int) int {
	// initial condition
	if n == 1 {
		return x
	}

	// even number
	if n%2 == 0 {
		return Pow(x*x, n/2)
	}

	// odd number
	return x * Pow(x*x, (n-1)/2)
}
